### ISSUE TITLE
Loki output plugin's auto_kubernetes_labels non-alphanumeric sanitization broken under load

### ISSUE THREAD

**Issue by tjorri** [fluent/fluent-bit#3700]
## Bug Report

**Describe the bug**
Using the Loki output plugin under moderate volumes with `auto_kubernetes_labels` causes the plugin to sometimes fail in sanitizing Loki label keys, causing non-alphanumeric characters not to be converted to underscores, which in turn leads to issues in retransmissions being out of order.

**To Reproduce**

Tested and was able to reproduce by running a fluent-bit pod in a Kubernetes cluster with Loki running in-cluster using the following fluent-bit configuration:

```
[SERVICE]
    Flush 1
    Log_Level info
    Daemon Off

[INPUT]
    Name dummy
    Tag dummy.log
    Rate 50
    Dummy {"log":"somelogentry","kubernetes":{"namespace_name":"default","pod_id":"31f023a3-be85-4be1-a4ca-a4e9ce9ca000","labels":{"label1":"value","label2":"value","label3":"value","label4":"value","label5":"value","complex.label/one":"foo","complex.label/two":"bar","complex.label/three":"value","complex.label/four":"value","complex.label/five":"value","complex.label/six":"value","complex.label/seven":"value","complex.label/eight":"value","complex.label/nine":"value","complex.label/ten":"value"},"annotations":{"kubernetes.io/psp":"eks.privileged","prometheus.io/port":"9090","prometheus.io/scrape":"true"},"host":"ip-10-0-0-10.eu-west-1.compute.internal","container_name":"server","docker_id":"81255dc58836a4b3c85baf49df6043054697fcf71a8d21e2029d31143e63f000","container_hash":"000011112222.dkr.ecr.eu-west-1.amazonaws.com/image@sha256:14b91dc338d08ba37df3a56bc79ac9aee9085396909c7bcdb06ac1c9ab8ed04e","container_image":"000011112222.dkr.ecr.eu-west-1.amazonaws.com/image:tag"}}

[OUTPUT]
    Name loki
    Match *
    host loki
    port 3100
    labels job=fluentbittest
    auto_kubernetes_labels on
    line_format json
```

**Expected behavior**
Expected behavior is for fluent-bit to successfully send the logs to Loki, converting the `complex.label/one` key to `complex_label_one`, and similarly for other keys.

**Screenshots**
```/ # /fluent-bit/bin/fluent-bit -c config
Fluent Bit v1.7.7
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/06/29 10:21:04] [Warning] [config] I cannot open /parsers.conf file
[2021/06/29 10:21:04] [ info] [engine] started (pid=115)
[2021/06/29 10:21:04] [ info] [storage] version=1.1.1, initializing...
[2021/06/29 10:21:04] [ info] [storage] in-memory
[2021/06/29 10:21:04] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2021/06/29 10:21:04] [ info] [output:loki:loki.0] configured, hostname=loki:3100
[2021/06/29 10:21:04] [ info] [sp] stream processor started
[2021/06/29 10:21:15] [error] [output:loki:loki.0] loki:3100, HTTP status=400
Error parsing labels '{complex.label/five="value", complex_label_eight="value", complex_label_four="value", complex_label_nine="value", complex_label_one="foo", complex_label_seven="value", complex_label_six="value", complex_label_ten="value", complex_label_three="value", complex_label_two="bar", job="fluentbittest", label1="value", label2="value", label3="value", label4="value", label5="value"}' with error: 1:9: parse error: unexpected character inside braces: '.'

[2021/06/29 10:21:15] [ warn] [engine] failed to flush chunk '115-1624962074.550207929.flb', retry in 8 seconds: task_id=0, input=dummy.0 > output=loki.0 (out_id=0)
[2021/06/29 10:21:23] [error] [output:loki:loki.0] loki:3100, HTTP status=400
entry with timestamp 2021-06-29 10:21:15.508237645 +0000 UTC ignored, reason: 'entry out of order' for stream: {complex_label_eight="value", complex_label_five="value", complex_label_four="value", complex_label_nine="value", complex_label_one="foo", complex_label_seven="value", complex_label_six="value", complex_label_ten="value", complex_label_three="value", complex_label_two="bar", job="fluentbittest", label1="value", label2="value", label3="value", label4="value", label5="value"},
total ignored: 1 out of 1

[2021/06/29 10:21:23] [ warn] [engine] chunk '115-1624962074.550207929.flb' cannot be retried: task_id=0, input=dummy.0 > output=loki.0
^C[2021/06/29 10:21:23] [engine] caught signal (SIGINT)
[2021/06/29 10:21:23] [ warn] [engine] service will stop in 5 seconds
[2021/06/29 10:21:23] [error] [output:loki:loki.0] loki:3100, HTTP status=400
Error parsing labels '{complex.label/five="value", complex_label_eight="value", complex_label_four="value", complex_label_nine="value", complex_label_one="foo", complex_label_seven="value", complex_label_six="value", complex_label_ten="value", complex_label_three="value", complex_label_two="bar", job="fluentbittest", label1="value", label2="value", label3="value", label4="value", label5="value"}' with error: 1:9: parse error: unexpected character inside braces: '.'

[2021/06/29 10:21:23] [ warn] [engine] failed to flush chunk '115-1624962075.553701678.flb', retry in 9 seconds: task_id=1, input=dummy.0 > output=loki.0 (out_id=0)
[2021/06/29 10:21:28] [ info] [engine] service stopped
```

**Your Environment**

* Version used: 1.7.7
* Configuration: See config above in "To Reproduce"
* Environment name and version (e.g. Kubernetes? What version?): AWS EKS 1.18.9, Loki 2.2.1
* Operating System and version: Run in Docker on EKS using fluent/fluent-bit:1.7.7-debug
* Filters and plugins: Loki

**Additional context**

* Turning `auto_kubernetes_labels` off will work as expected with Loki ingesting the logs without issues as the label key conversion doesn't fail.
* This issue seems to be slightly sporadic, e.g. a bulk of the dummy tests go through, but occasionally this fails in the way described above.
* Went ahead and tested this with fluent/fluent-bit:1.7.9-debug as well, and the same issue persists.

**Comment by Kampe** [comment]
seeing the same issue with loki 2.2.1

**Comment by richardburakowski** [comment]
fyi, the `Error parsing labels` will be a symptom fixed in #3839
